### PR TITLE
Feature/refactor node chat

### PR DESCRIPTION
**Pull request by ProbablePrime**
A few fixes to our chat docs.

### CODE REVIEW

File: build/misc.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ function getLocals () {
     const out = {
         _,
         orderObject,
+        log: console.log,
```
Comment (SimonSchick) [inline build/misc.js]: leftover debug?

File: src/tutorials/code/node/chatbot/2.js [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+});
+
+
+/**
+ * Creates a beam chat socket and sets up listeners to various chat events.
+ * @param {any} userId The user to authenticate as
+ * @param {any} channelId The channel id to join
+ * @param {any} endpoints An endpoints array from a beam.chat.join call.
+ * @param {any} authkey An authentication key from a beam.chat.join call.
+ * @returns Promise.<>
```
Comment (SimonSchick) [inline src/tutorials/code/node/chatbot/2.js]: Should be wrapped in `{}`.

File: build/misc.js (hunk 1 above)
Comment (ProbablePrime) [inline build/misc.js]: Intentional, It does no harm being there and my local copy keeps getting it deleted by stash and branch operations.

File: src/tutorials/chatbot.pug [hunk 3]
```diff
@@ -43,13 +43,15 @@ block tutorialContent
                 The result will look something like this in it you can see an array of chat servers that we can connect to within the #[code endpoint] array.:
             +highlightFile('javascript', 'tutorials/code/node/chatbot/1_response.txt')
             p.
-                Next we will use this response to connect to the chat server, for that we have the #[code createChatSocket()] function. This takes the information from the response and creates a Beam chat socket from it.
+                Next we will use this response to connect to the chat server,
+                Let's write the #[code createChatSocket()] function and uncomment its use in the previous code.
```
Comment (connor4312) [inline src/tutorials/chatbot.pug]: no wrapping!

File: src/tutorials/code/node/chatbot/1.js [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
+    // Store the logged in user's details for later refernece
     userInfo = response.body;
+    // Returns a promise that resolves with our chat connection details.
     return client.chat.join(response.body.channel.id);
 })
 .then(response => {
     const body = response.body;
     console.log(body);
-    return createChatSocket(userInfo.id, userInfo.channel.id, body.endpoints, body.authkey);
+    // return createChatSocket(userInfo.id, userInfo.channel.id, body.endpoints, body.authkey);
```
Comment (connor4312) [inline src/tutorials/code/node/chatbot/1.js]: intentional? Maybe replacing it with `// todo: connect to chat` would be nicer

File: src/tutorials/code/node/chatbot/3.js [hunk 5]
```diff
@@ -15,8 +15,17 @@ client.use('password', {
...
+    const body = response.body;
+    return createChatSocket(userInfo.id, userInfo.channel.id, body.endpoints, body.authkey);
+})
+.catch(error => {
+    console.log('Something went wrong:', error);
+});
+
+
+function createChatSocket (userId, channelId, endpoints, authkey) {
+      // Chat connection
```
Comment (connor4312) [inline src/tutorials/code/node/chatbot/3.js]: Weird intentation

**Comment by ProbablePrime** [comment]
Sorry for the delay here I lost this PR.

@connor4312 @SimonSchick updated

**Pull request merged**